### PR TITLE
A few more Genus 2 tweaks

### PR DESCRIPTION
**Pull request by AndrewVSutherland**
This PR makes a few very minor formatting changes based on feedback I have received.  Rational points are now listed in order by naive height (using L1 norm and then lex-order to break ties), factors of (-1) are removed from the factorization of divisors, and the indentation/spacing has been tweaked in several places

### CODE REVIEW

**Comment by AndrewVSutherland** [comment]
Essentially all the changes are visible on this page:

https://red.lmfdb.xyz/Genus2Curve/Q/758059/a/758059/1
vs
https://beta.lmfdb.org/Genus2Curve/Q/758059/a/758059/1

The one other change is fixing a broken piece of html that prevented the endomorphism algebra from being properly displayed on isogeny classes, see:

https://red.lmfdb.xyz/Genus2Curve/Q/758059/a/
vs
https://beta.lmfdb.org/Genus2Curve/Q/758059/a/

**Comment by roed314** [comment]
I can take a look in a couple hours.

**Comment by AndrewVSutherland** [comment]
@roed314 thanks,  I'll stop fiddling in 30 minutes (I'm editing the knowls but occasional notice tiny formatting issues, none of which are critical so feel free to merge whenever).

File: lmfdb/genus2_curves/web_g2c.py [hunk 1]
```diff
@@ -430,9 +431,9 @@ def split_statement(coeffs, labels, condnorms):
             statement += """<br>&nbsp;&nbsp;Elliptic curve <a href=%s>%s</a>""" % (url_for_ec(label), label)
         # Otherwise give defining equation:
         else:
-            statement += """<br>&nbsp;&nbsp;\(y^2 = x^3 - g_4 / 48 x - g_6 / 864\) with<br>\
-            \(g_4 = %s\)<br>\
-            \(g_6 = %s\)<br>\
+            statement += """<br>&nbsp;&nbsp;\\(y^2 = x^3 - g_4 / 48 x - g_6 / 864\\) with<br>\
+            \\(g_4 = %s\\)<br>\
```
Comment (roed314) [inline lmfdb/genus2_curves/web_g2c.py]: Shouldn't these trailing backslashes be removed?  It seems like you intended Python line continuation backslashes, but they're not necessary in HTML.

File: lmfdb/genus2_curves/web_g2c.py [hunk 2]
```diff
@@ -534,7 +537,20 @@ def local_table(D,N,tama,bad_lpolys):
     loctab.extend(['</tbody>', '</table>'])
     return '\n'.join(loctab)
 
+def ptscmp(a,b):
```
Comment (roed314) [inline lmfdb/genus2_curves/web_g2c.py]: This is not Python 3 compatible.  See "The cmp Argument" section of https://portingguide.readthedocs.io/en/latest/comparisons.html.

**Comment by roed314** [comment]
If I'm happy with the PR (see my comments) and merge it after you go to sleep, should I push to dev and web?  Or wait for you to do it in the morning?  I don't know if you're going to have time to address my comments tonight.

File: lmfdb/genus2_curves/web_g2c.py (hunk 1 above)
Comment (AndrewVSutherland) [reply lmfdb/genus2_curves/web_g2c.py]: Yes (this has been like this for ages, not changed in this PR, but I agree it should be cleaned up and the indentation is wrong in any case). I just fixed this.

File: lmfdb/genus2_curves/web_g2c.py (hunk 2 above)
Comment (AndrewVSutherland) [reply lmfdb/genus2_curves/web_g2c.py]: Converted cmp to key

Comment (AndrewVSutherland) [reply lmfdb/genus2_curves/web_g2c.py]: You can see this in action on
https://red.lmfdb.xyz/Genus2Curve/Q/792079/a/792079/1
(or any curve with lots of rational points)

**Comment by AndrewVSutherland** [comment]
If you can merge to master (once your happy) that's fine, I can push to dev first thing in the morning.  The push to web needs to wait until knowl review finishes (@jvoight and @JohnCremona are helping with this), because there are previously reviewed knowls with changes related to this code change that won't be visible on web until they are marked as reviewed -- I'll make sure all the essential ones get wrapped up tomorrow (one way or another) and then push to web.

File: lmfdb/genus2_curves/web_g2c.py (hunk 1 above)
Comment (AndrewVSutherland) [reply lmfdb/genus2_curves/web_g2c.py]: You can see an example running the modified code at:
https://red.lmfdb.xyz/Genus2Curve/Q/999936/a/999936/1

**Comment by roed314** [comment]
Should I hold off on #3620 until you push to web then?

**Comment by AndrewVSutherland** [comment]
@roed314 Yeah, just to be safe, let's hold off on merging #3620 until after this PR has been merged and pushed to web.

**Comment by roed314** [comment]
Ok, looks good.

**Pull request merged**